config/config_mgmt.py

```diff
@@ -0,0 +1,709 @@
+    In: Config to be searched
+    skeys: Keys to be searched in In Config i.e. search Keys.
+    Out: Contains the search result
+    """
+    def _searchKeysInConfig(self, In, Out, skeys):
+
+        found = False
+        if isinstance(In, dict):
+            for key in In.keys():
+                #print("key:" + key)
+                for skey in skeys:
+                    # pattern is very specific to current primary keys in
+                    # config DB, may need to be updated later.
+                    pattern = '^' + skey + '\|' + '|' + skey + '$' + \
+                        '|' + '^' + skey + '$'
+                    #print(pattern)
+                    reg = re.compile(pattern)
+                    #print(reg)
+                    if reg.search(key):
+                        # In primary key, only 1 match can be found, so return
+                        # print("Added key:" + key)
+                        Out[key] = In[key]
+                        found = True
+                        break
+                # Put the key in Out by default, if not added already.
+                # Remove later, if subelements does not contain any port.
+                if Out.get(key) is None:
+                    Out[key] = type(In[key])()
+                    if self._searchKeysInConfig(In[key], Out[key], skeys) == False:
+                        del Out[key]
+                    else:
+                        found = True
+
+        elif isinstance(In, list):
+            for skey in skeys:
+                if skey in In:
+                    found = True
+                    Out.append(skey)
+                    #print("Added in list:" + port)
+
+        else:
+            # nothing for other keys
+            pass
+
+        return found
+
+    """
+    This function returns the relavant keys in Input Config.
+    For Example: All Ports related Config in Config DB.
+    """
+    def configWithKeys(self, configIn=dict(), keys=list()):
+
+        configOut = dict()
+        try:
+            if len(configIn) and len(keys):
+                self._searchKeysInConfig(configIn, configOut, skeys=keys)
+        except Exception as e:
+            print("configWithKeys Failed, Error: {}".format(str(e)))
+            raise e
+
+        return configOut
+
+    """
+    Create a defConfig for given Ports from Default Config File.
+    """
+    def _getDefaultConfig(self, ports=list()):
+
+        # function code
+        try:
+            print("Generating default config for {}".format(ports))
+            defConfigIn = readJsonFile(DEFAULT_CONFIG_DB_JSON_FILE)
+            #print(defConfigIn)
+            defConfigOut = dict()
+            self._searchKeysInConfig(defConfigIn, defConfigOut, skeys=ports)
+        except Exception as e:
+            print("getDefaultConfig Failed, Error: {}".format(str(e)))
+            raise e
+
+        return defConfigOut
+
+    def _updateDiffConfigDB(self):
+
+        # main code starts here
```

Comment:
This comment seems out of place. Left over from when the code had a main() function?

Reply:
_updateDiffConfigDB earlier had an inner function, so I had a comment to point where the main code for the function starts. I will clean it up now.

Reply:
Done.